mappings/net/minecraft/data/server/recipe/SmithingRecipeJsonBuilder.mapping

```diff
@@ -9,11 +9,11 @@ CLASS net/minecraft/class_5377 net/minecraft/data/server/recipe/SmithingRecipeJs
 		ARG 2 base
 		ARG 3 addition
 		ARG 4 result
-	METHOD method_29729 create (Lnet/minecraft/class_1856;Lnet/minecraft/class_1856;Lnet/minecraft/class_1792;)Lnet/minecraft/class_5377;
+	METHOD method_29729 createRecipe (Lnet/minecraft/class_1856;Lnet/minecraft/class_1856;Lnet/minecraft/class_1792;)Lnet/minecraft/class_5377;
```

Comment:
Let's loot at this.

## The class
First, this class is actually a builder, even though its fields are all final. So the rename looks good.

## The method

This method returns a new instance of this class. Your current name `createRecipe` is really weird, as it sounds like creating a recipe than creating a builder of/for/from a recipe. Consider this code:
```java
SmithingRecipeJsonBuilder builder = SmithingRecipeJsonBuilder.createRecipe(/* */);
```
Am I creating a builder or a recipe?

I suggest a name like `forRecipe`, `ofRecipe`, `of`, or `create` instead. They are all less confusing.

Reply:
`createRecipe` is just something I quickly came up with that was less confusing than `SmithingRecipeJsonFactory::create`, it probably isn't a lot more descriptive. Thinking about it more, I think any kind of `create` is pretty confusing in `Builder`s to begin with, it might make sense to have a `static Thing ThingBuilder#create` in a `Builder` when it doubles as a `Factory`, but when it comes to `ThingBuilder ThingBuilder::create`, at the very least `ThingBuilder::begin`, but there is almost always a more descriptive name to come up with. I like the `output` suggestion further below. Perhaps `ThingBuilder::forOutput`, to distinguish the static methods from the rest.

Reply:
I don't like `forRecipe`, in that case I prefer `recipeFor` since it's how you would write it in english.

Reply:
See also #1939 for how choosing a name here might affect mappings elsewhere.